tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -73,5 +75,79 @@ internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, s
 
             return string.Empty;
         }
+
+        /// <summary>
+        /// Uses a sql instruction to set a context for the current connection, bearing the span ID and trace ID.
+        /// This is meant to circumvent cache invalidation issues that occur when those values are injected in comment.
+        /// Currently only working for MSSQL (uses an instruction that is specific to it)
+        /// </summary>
+        /// <returns>True if the traceparent information was set</returns>
+        internal static bool PropagateDataViaContext(Tracer tracer, DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbConnection? connection, string serviceName, Scope scope, SqlTags tags)
+        {
+            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || connection == null)
+            {
+                return false;
+            }
+
+            // we want the instrumentation span to be a sibling of the actual query span
+            var instrumentationParent = scope.Parent?.Span?.Context;
+            // copy the tags so that modifications on one span don't impact the other
+            var copyProcessor = new ITags.CopyProcessor<SqlTags>();
+            tags.EnumerateTags(ref copyProcessor);
+            using (var instrumentationScope = tracer.StartActiveInternal("set context_info", instrumentationParent, tags: copyProcessor.TagsCopy, serviceName: serviceName))
```

Comment:
Do we want the operation name to be `"set context_info"`? That looks more like a resource name to me.

Reply:
Hmm, yeah, the operation would be something like "statement instrumentation" for instance ?
I know it's not a good argument but I took the same operation name as [what was done in Java](https://github.com/DataDog/dd-trace-java/blob/c3e9fa9bab03b39fc8e33ecacdfb8b127eb958e8/dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/JDBCDecorator.java#L267)

Reply:
Operation names usually look like `{library}.{action}`
```
aspnetcore.request
mysql.query
http.client.request
redis.command
```

I don't know what to use here, maybe something generic like `dd.instrumentation` or specific to this case like `sql-server.set-context` (since the other span will be `sql-server.query`).